spec/features/admin/variants_spec.rb

```diff
@@ -7,24 +7,98 @@
+      login_to_admin_section
+      visit spree.admin_product_variants_path product
+      click_link 'New Variant'
+
+      fill_in 'unit_value_human', with: '1'
+      fill_in 'variant_unit_description', with: 'foo'
+      click_button 'Create'
+
+      # Then the variant should have been created
+      expect(page).to have_content "Variant \"#{product.name}\" has been successfully created!"
+    end
+
+    scenario "creating a new variant from product variant page with filter" do
+      # Given a product with a unit-related option type
+      product = create(:simple_product, variant_unit: "weight", variant_unit_scale: "1")
+      filter = { producerFilter: 2 }
+
+      login_to_admin_section
+
+      visit spree.admin_product_variants_path(product, filter)
+      click_link 'New Variant'
+
+      uri = URI.parse(current_url)
+      expect("#{uri.path}?#{uri.query}").to eq spree.new_admin_product_variant_path(product, filter)
+
+      # Cancel link should include product filter
+      expected_cancel_url = Regexp.new(
+        Regexp.escape(spree.admin_product_variants_path(product, filter))
+      )
+      expect(page).to have_link(I18n.t('actions.cancel'), href: expected_cancel_url)
+    end
+  end
 
-    fill_in 'unit_value_human', with: '1'
-    fill_in 'variant_unit_description', with: 'foo'
-    click_button 'Create'
+  describe "viewing product variant" do
+    scenario "when the product page has a product filter" do
+      # Given a product with a unit-related option type
+      product = create(:simple_product, variant_unit: "weight", variant_unit_scale: "1")
+      filter = { producerFilter: 2 }
+
+      # When I create a variant on the product
+      login_to_admin_section
+
+      visit spree.admin_product_variants_path(product, filter)
+
+      expected_new_url = Regexp.new(
+        Regexp.escape(spree.new_admin_product_variant_path(product, filter))
+      )
+      expect(page).to have_link("New Variant", href: expected_new_url)
+
+      expected_show_delete_url = Regexp.new(
+        Regexp.escape(spree.admin_product_variants_path(product, { deleted: 'on' }.merge(filter)))
+      )
+      expect(page).to have_link("Show Deleted", href: expected_show_delete_url)
+
+      # Variant link should include product filter
+      variant = product.variants.first
+
+      expected_edit_url = Regexp.new(
+        Regexp.escape(spree.edit_admin_product_variant_path(product, variant, filter))
+      )
+      expect(page).to have_link(I18n.t(:edit), href: expected_edit_url)
 
-    # Then the variant should have been created
-    expect(page).to have_content "Variant \"#{product.name}\" has been successfully created!"
+      expected_delete_url = Regexp.new(
+        Regexp.escape(spree.admin_product_variant_path(product, variant, filter))
+      )
+      expect(page).to have_link(I18n.t(:delete), href: expected_delete_url)
+    end
   end
 
   describe "editing unit value and description for a variant", js: true do
```

Comment:
there's already a js:true at the top of the spec so this one is not needed, right?

Reply:
If I am not mistaken this "describe" sits under (top of the file) : 
```
feature '
    As an admin
    I want to manage product variants
' do
```
and not under any other "describe", so I think we need the js:true here